pkg/manifests/manifests.go

```diff
@@ -78,17 +79,26 @@ func (f *Factory) RouterDaemonSet(cr *ingressv1alpha1.ClusterIngress) (*appsv1.D
 
 	ds.Name = name
 
-	ds.Labels = map[string]string{
-		"app": "router",
-	}
+	ds.Spec.Template.Labels["router"] = name
 
 	ds.Spec.Selector = &metav1.LabelSelector{
 		MatchLabels: map[string]string{
-			"app": name,
+			"app":    "router",
+			"router": name,
 		},
 	}
 
-	ds.Spec.Template.Labels["app"] = name
+	routeSelector, err := metav1.LabelSelectorAsSelector(cr.Spec.RouteSelector)
+	if err != nil {
+		return nil, fmt.Errorf("clusteringress %q has invalid spec.routeSelector: %v", cr.Name, err)
+	}
+	env := []corev1.EnvVar{
+		{Name: "ROUTER_SERVICE_NAME", Value: cr.Name},
+		{Name: "ROUTER_SERVICE_NAMESPACE", Value: ds.Namespace},
+		{Name: "ROUTER_CANONICAL_HOSTNAME", Value: cr.Spec.IngressDomain},
+		{Name: "ROUTE_LABELS", Value: routeSelector.String()},
```

Comment:
The `oc adm router` command sets `ROUTE_LABELS` only if the selector is nonempty.  It looks like the router handles an empty value fine though, so it's probably not worth bothering to conditionalize exporting the variable.

Reply:
I made this and canonical hostname optionally assigned